src/include/sof/audio/format.h

```diff
@@ -79,6 +79,13 @@
 #define SATP_INT32(x) (((x) > INT32_MAX) ? INT32_MAX : (x))
 #define SATM_INT32(x) (((x) < INT32_MIN) ? INT32_MIN : (x))
 
+/* Arithmetic shift right */
+#define ARITHMETICSHIFTRIGHT(u, n) ({	\
```

Comment:
How does this macro differ from C language shift ">>" for signed integer type? It's arithmetic shift.

Reply:
Yes, it's arithmetic shift

 Arithmetic Shift Right operates on both unsigned and signed inputs. If the input is unsigned, asr shifts zeros into the
 positions of bits that it shifts right. If the input is signed, asr shifts the MSB into the positions
 of bits that it shifts right.

right shifts on signed integers first casts the signed integers to unsigned integers, and then right shifts the unsigned integers, Using an arithmetic shift, the sign bit would be kept.